src/app/clinic/clinic_facade.cpp

```diff
@@ -88,6 +104,14 @@ void Clinic::removeVisit(const std::shared_ptr<Visit>& visit)
         {
             (*visit_it)->getDoctor().lock()->removeVisit(*visit_it);
         }
```

Comment:
I think that weak_ptr::expired calls lock() and if(locked_ptr) inside, so this might be less optimal.
And it produces less readable code. Consider for example:
```
auto visits_doctor = (*visit_it)->getDoctor().lock()
if (doctor)
{
    visits_doctor->->removeVisit(*visit_it);
}
```
another two if could be refactored in the same manner.

Reply:
appendTempVisit() --> setTempVisit()